lms/djangoapps/lms_xblock/mixin.py

```diff
@@ -131,7 +133,7 @@ def _get_user_partition(self, user_partition_id):
         Returns the user partition with the specified id.  Raises
         `NoSuchUserPartitionError` if the lookup fails.
         """
-        for user_partition in self.user_partitions:
+        for user_partition in self.runtime.service(self, 'partitions').course_partitions:
```

Comment:
@nasthagiri this method is what the access control code calls. Previously, "self" was any xblock, not just the course. I am now looking at on the course user_partitions, but I think that is fine since that is the only place they should be modified. 

For reference, here is the access usage: https://github.com/edx/edx-platform/blob/b287f48f8b9ea31e2ad42be8f609dbb0e0059ef9/lms/djangoapps/courseware/access.py#L496